idaes/gas_distribution/unit_models/tests/test_compressor.py

```diff
@@ -0,0 +1,273 @@
+            n_scen,
+            input_values,
+            output_values=target_values,
+            to_fix=input_values,
+        )
+        with param_sweeper:
+            self.assertEqual(degrees_of_freedom(m), 0)
+            for inputs, target in param_sweeper:
+                ipopt.solve(m, tee=False)
+
+                # Sanity check that inputs have been properly set.
+                for var, val in inputs.items():
+                    val = pyo.value(pyo.units.convert(val, var.get_units()))
+                    self.assertEqual(val, var.value)
+
+                # Check that calculated power requirements are close to what
+                # we expect. Values are different due to discrepancy in cv
+                # between paper and IDAES.
+                for var, val in target.items():
+                    val = pyo.value(pyo.units.convert(val, var.get_units()))
+                    #self.assertAlmostEqual(val, var.value, delta=1e2)
+                    #self.assertAlmostEqual(val, var.value, reltol=0.1)
+                    # Forget the relative tolerance argument for
+                    # assertAlmostEqual, so using pytest.approx for now...
+                    self.assertEqual(val, pytest.approx(var.value, rel=0.1))
+                    # We use a pretty generous margin here because the
+                    # property package's heat capacity cv is different
+                    # from that used in the paper. (Paper does not use
+                    # exactly (cp - R))
+
+
+@pytest.mark.unit
+class TestSimpleCompressor(unittest.TestCase):
+
+    def test_compressor_methane(self):
+        """
+        Test constructing the compressor with a simple methane
+        property package.
+        """
+        m = pyo.ConcreteModel()
+        default = {
+            "dynamic": False,
+        }
+        m.fs = idaes.FlowsheetBlock(default=default)
+        m.fs.properties = NaturalGasParameterBlock()
+        compressor_config = {
+            "property_package": m.fs.properties,
+        }
+        m.fs.compressor = IsothermalCompressor(default=compressor_config)
+
+        m.fs.compressor.inlet_state[0].temperature.fix(300.0*pyo.units.K)
+        m.fs.compressor.inlet_state[0].pressure.fix(20.0*pyo.units.bar)
+        m.fs.compressor.boost_pressure[0].fix(10.0*pyo.units.bar)
+        m.fs.compressor.inlet_state[0].flow_mol.fix(1000.0)
+        # Fix mole_frac as defined_state is True at inlet
+        j = next(iter(m.fs.properties.component_list))
+        m.fs.compressor.inlet_state[0].mole_frac_comp[j].fix(1.0)
+
+        # This test asserts:
+        # (a) consistent units
+        # (b) zero degrees of freedom
+        # (c) a perfect matching between constraints and variables
+        assert_units_consistent(m)
+
+        igraph = IncidenceGraphInterface(m)
+        matching = igraph.maximum_matching()
+        var_dmp, con_dmp = igraph.dulmage_mendelsohn()
+        N, M = igraph.incidence_matrix.shape
```

Comment:
Is this used? It appears to be repeated with an assert statement below.

Reply:
Yep, I'll remove this.